src/environmentd/src/deployment/preflight.rs

```diff
@@ -154,34 +155,84 @@ pub async fn preflight_0dt(
     if catalog_generation < deploy_generation {
         info!("this deployment is a new generation; booting in read only mode");
 
-        let (caught_up_trigger, caught_up_receiver) = trigger::channel();
+        let (caught_up_trigger, mut caught_up_receiver) = trigger::channel();
 
         // Spawn a background task to handle promotion to leader.
         mz_ore::task::spawn(|| "preflight_0dt", async move {
-            info!("waiting for deployment to be caught up");
+            let (initial_next_user_item_id, initial_next_replica_id) = get_next_ids(
+                boot_ts,
+                persist_client.clone(),
+                environment_id.clone(),
+                deploy_generation,
+                Arc::clone(&catalog_metrics),
+                bootstrap_args.clone(),
+            )
+            .await;
+
+            info!(
+                %initial_next_user_item_id,
+                %initial_next_replica_id,
+                "waiting for deployment to be caught up");
 
             let caught_up_max_wait_fut = async {
                 tokio::time::sleep(caught_up_max_wait).await;
                 ()
             };
+            let mut caught_up_max_wait_fut = pin!(caught_up_max_wait_fut);
 
-            let skip_catchup = deployment_state.set_catching_up();
+            let mut skip_catchup = deployment_state.set_catching_up();
 
-            tokio::select! {
-                () = caught_up_receiver => {
-                    info!("deployment caught up");
-                }
-                () = skip_catchup => {
-                    info!("skipping waiting for deployment to catch up due to administrator request");
-                }
-                () = caught_up_max_wait_fut => {
-                    if panic_after_timeout {
-                        panic!("not caught up within {:?}", caught_up_max_wait);
+            let mut check_ddl_changes_interval = tokio::time::interval(Duration::from_secs(5 * 60));
```

Comment:
Is it possible to make this configurable via LaunchDarkly? Alternatively could we make this an optional CLI arg that can be provided to `environmentd`. Having a knob here to tune this seems useful